src/main/scala/org/apache/spark/sql/hive/llap/LlapContext.scala

```diff
@@ -124,6 +122,44 @@ class LlapContext(sc: SparkContext,
       Seq.empty
     }
   }
+
+  override protected[sql] def executePlan(plan: LogicalPlan): super.QueryExecution = {
+    new this.QueryExecution(plan)
+    super.executePlan(plan)
+  }
+
+  private[sql] class QueryExecution(logicalPlan: LogicalPlan)
+    extends super.QueryExecution(logicalPlan)  {
+
+    /**
+     * Access control for Desc Table Function
+     */
+    val checkDesc = executedPlan match {
+      case ExecutedCommand(desc: DescribeCommand) =>
+        var database = ""
+        var tablename = ""
+        val args = desc.argString
+        args.split(",").foreach{
+          s =>
+            if (s.contains("table"))
+              {
+                val tables = s.split("->")
+                val nameParts = tables{1}.split("\\.")
+                if (nameParts.length != 2) {
```

Comment:
0: jdbc:hive2://ctr-e133-1493418528701-160107> use spark_ranger_test;
+---------+--+
| result  |
+---------+--+
+---------+--+
No rows selected (1.272 seconds)
0: jdbc:hive2://ctr-e133-1493418528701-160107> desc t_no;
Error: org.apache.spark.sql.AnalysisException: Table not found: t_no; line 1 pos 5 (state=,code=0)

I think this might be the limit of spark 1.6.x. ?